src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -533,6 +556,27 @@ public void startJenkins() throws Throwable {
         }
     }
 
+    private String getWarAbsolutePath() throws Exception {
+        return war == null ? findJenkinsWar().getAbsolutePath() : war.getAbsolutePath();
+    }
+
+    private static boolean supportsPortFileName(String war) throws IOException {
+        try (JarFile warFile = new JarFile(war)) {
```

Comment:
Is this called multiple times per test? If so, can we cache t? Might be slow on Windows.

Reply:
I can probably optimize it.